pkg/controllers/provisioningv2/provisioningcluster/controller.go

```diff
@@ -67,10 +67,11 @@ func Register(ctx context.Context, clients *wrangler.Context) {
 
 	if features.MCM.Enabled() {
 		h.dynamicSchema = clients.Mgmt.DynamicSchema().Cache()
-		h.mgmtClusterCache = clients.Mgmt.Cluster().Cache()
-		h.mgmtClusterClient = clients.Mgmt.Cluster()
 	}
 
+	h.mgmtClusterCache = clients.Mgmt.Cluster().Cache()
```

Comment:
Can you move these up to ~line 65?

Reply:
Of course, Thanks for highlighting that!